go/consensus/tendermint/apps/staking/auth.go

```diff
@@ -62,6 +62,14 @@ func (app *stakingApplication) PostExecuteTx(ctx *api.Context, tx *transaction.T
 		return transaction.ErrInsufficientFeeBalance
 	}
 
+	params, err := state.ConsensusParameters(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to fetch staking consensus parameters: %w", err)
+	}
+	if account.General.Balance.Cmp(&params.MinTransactBalance) < 0 {
+		return staking.ErrUnderMinTransferAmount
```

Comment:
adding this check at the end has two purposes:
1. prevent users from wasting accounts moving the minimum amount through many accounts
2. possibly be nice and prevent someone from accidentally locking themselves out by delegating everything

but there are problems
1. this is probably the wrong place to put it. we don't want to revert the fee payment because by now we've done the work. but paying the fee might be the thing that puts the account below the limit. could we determine this in general before doing the work? that's hard to know though, with a transfer additionally lowering the balance or a withdrawal increasing the balance.
2. other non-transaction things can bring you below the minimum, such as an allowed account withdrawing

Reply:
Should withdrawals that would bring the account below the minimum be rejected?

Reply:
alright let's try this:

- no more postexecute check
- any action that reduce an account balance must pinky promise to check for minimum balance :skull: 
  - including fee payment
  - including transfer from an account by a transaction
  - including someone withdrawing from an account
  - including a paratime transferring the last of its balance, even though that doesn't leave a nonce behind
- certain infrastructural balances are exempt
  - common pool
  - fee accumulator

some things that this entails:

- in order to withdraw, you must already have min balance + fee already. doesn't matter if the withdrawal would have covered it
- a paratime can't transfer away the last of its balance either, even though this wouldn't leave a nonce behind
- you can still transfer a less-than-minimum amount to a new account. probably will have to combine this with the minimum transfer amount feature

Reply:
too complicated for this timeframe. going with the post-withdrawal check :+1: